### PR TITLE
Frozen index input clone copy cache file

### PR DESCRIPTION
**Pull request by henningandersen**
Frozen index input clone now copies the cache file object ensuring that the fast path optimizations work.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/store/input/FrozenIndexInputTests.java [hunk 1]
```diff
@@ -83,7 +85,7 @@ public void testRandomReads() throws IOException {
             cacheSize = ByteSizeValue.ofBytes(randomLongBetween(1L, 10L) * regionSize.getBytes() + randomIntBetween(0, 100));
         }
 
-        final Settings settings = Settings.builder()
+        final Settings settings1 = Settings.builder()
```
Comment (original-brownbear) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/store/input/FrozenIndexInputTests.java:88]: What am I missing here, why do we need that redundant variable?

**Review by original-brownbear — APPROVED** [review]
LGTM, just a small cleanup in the test needed unless I'm missing something :) Thanks!

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/store/input/FrozenIndexInputTests.java (hunk 1 above)
Comment (henningandersen) [reply x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/store/input/FrozenIndexInputTests.java:88]: Thanks, that makes no sense, is a leftover from refactoring this in an earlier iteration on it.

**Comment by henningandersen** [comment]
Thanks Armin!

**Pull request merged**